packages/flutter/lib/src/material/carousel.dart

```diff
@@ -480,6 +523,7 @@ class _CarouselViewState extends State<CarouselView> {
     if (widget.controller == null) {
       _internalController = CarouselController();
     }
+    _lastReportedLeadingIndex = _controller.initialItem;
```

Comment:
The definitions of leading item and initial item are different. So this initialization might not be accurate.

Leading item is just the first (partially) visible item for both fixed-weight carousel and weighted carousel.
Initial item is the first visible item for fixed-weight carousel but is the first primary item for weighted carousel.

I would introduce another `initialLeadingItem` in the controller class and then set `_lastReportedLeadingIndex = initialLeadingItem`:)